src/token-provider.ts

```diff
@@ -1,46 +1,52 @@
+import { RetryStrategy } from './retry-strategy';
 export interface TokenProvider {
     fetchToken(): Promise<string>;
+    invalidateToken(token?: string);
 }
 
-export class SimpleTokenProvider implements TokenProvider {
-    constructor(public jwt: string) { }
-    fetchToken(): Promise<string> {
-        return new Promise<string>((resolve, reject) => {
-            resolve(this.jwt);
-        });
+/**
+ * Wrapper around the token provider that contains a retry strategy
+ * TODO: test
+ */
+export class RetryingTokenProvider implements TokenProvider {
+    constructor(
+        private baseTokenProvider: TokenProvider, 
+        private retryStrategy: RetryStrategy){}
+
+    fetchToken(){
+        return this.tryFetching();
+    }
+
+    invalidateToken(token?: string){
```

Comment:
Ah I see this is presumably where tokens are removed from caches, etc. 

Nothing seems to ever call it though, unless I'm missing it?